src/test/java/uk/gov/hmcts/reform/wastandalonetaskbpmn/bpmn/CamundaCreateTaskTest.java

```diff
@@ -18,4 +27,76 @@ public void should_create_a_task_with_no_delay_until() {
         createTask(false);
     }
 
+    @Test
+    public void should_not_create_a_task_with_different_TenantId_multiple_resource() {
+
+        clearDeployments();
+
+        RepositoryService repositoryService = processEngineRule.getRepositoryService();
+
+        DeploymentBuilder deploymentBuilder = repositoryService.createDeployment()
+            .tenantId("someTenantId");
+        deploymentBuilder.addClasspathResource("wa-task-initiation-ia-asylum.bpmn")
+            .deploy();
+
+        DeploymentBuilder deploymentBuilder2 = repositoryService.createDeployment()
+            .tenantId("someTenantId2");
+
+        deploymentBuilder2.addClasspathResource("wa-task-initiation-ia-asylum_duplicate.bpmn")
+            .deploy();
+
+        assertThrows(MismatchingMessageCorrelationException.class, () -> {
+            createTask(false);
+        });
+    }
+
+    @Test
+    public void should_create_a_task_with_and_without_TenantId_multiple_resources() {
+
+        clearDeployments();
+
+        RepositoryService repositoryService = processEngineRule.getRepositoryService();
+
+        DeploymentBuilder deploymentBuilder = repositoryService.createDeployment()
+            .tenantId("someTenantId");
+        deploymentBuilder.addClasspathResource("wa-task-initiation-ia-asylum.bpmn")
+            .deploy();
+
+        DeploymentBuilder deploymentBuilder2 = repositoryService.createDeployment();
+
+        deploymentBuilder2.addClasspathResource("wa-task-initiation-ia-asylum_duplicate.bpmn")
+            .deploy();
+
+        ProcessInstance processInstance =  createTask(false);
+
+        BpmnAwareTests.complete(BpmnAwareTests.task("processTask"));
+        BpmnAwareTests.assertThat(processInstance).isEnded();
```

Comment:
It's a bit unclear which process this is creating a Task in since you have 2 tenantIds with identical steps.

Reply:
Actually Ravi, here one of the BPMN has tenantId and the other one has no tenantId. createTask() method is configured to look for BPMN without tenantId. So its fetching the BPMN without the tenantId and not failing.

Reply:
I think the test should check you've received the correct process e.g. the tenantId / it's absence or deployment key.

Reply:
I added assert statement for not having tenantId for the test Ravi.